addons/project_timesheet_holidays/models/resource.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+from collections import defaultdict
+from pytz import utc
+
+class ResourceCalendarLeaves(models.Model):
+    _inherit = "resource.calendar.leaves"
+
+    timesheet_ids = fields.One2many('account.analytic.line', 'global_leave_id', string="Analytic Lines")
+
+    def _work_time_per_day(self):
+        self.ensure_one()
+        if not self.date_from.tzinfo:
+            date_from = self.date_from.replace(tzinfo=utc)
+        if not self.date_to.tzinfo:
+            date_to = self.date_to.replace(tzinfo=utc)
```

Comment:
Could you try this to check if it solves the issue from the tester ? 
```suggestion
tz = pytz.timezone(self.calendar_id.tz)
date_from = tz.localize(self.date_from)
date_to = tz.localize(self.date_to)
```